### PR TITLE
fix: path display error when start with `crate`

### LINKED ISSUE #10172: Missing `::` on editor hover tip
See this screenshot:

hover

I expect the hover tip shows `fn foo() -> crate::X`, but I got `fn foo() -> crateX`, the `::` is missing.

### PR DESCRIPTION
**Pull request by dzvon**
Fixes  #10172

### CODE REVIEW

File: crates/hir_ty/src/display.rs [hunk 1]
```diff
@@ -1154,7 +1154,7 @@ impl HirDisplay for Path {
         }
 
         for (seg_idx, segment) in self.segments().iter().enumerate() {
-            if seg_idx != 0 {
+            if seg_idx != 0 || matches!(self.kind(), PathKind::Crate) {
```
Comment (flodiebold) [inline crates/hir_ty/src/display.rs:1157]: it seems like `super` and `self` would have the same problems?

Comment (dzvon) [reply crates/hir_ty/src/display.rs]: I knew, I was just fixing it. It should be okay for now.

I'll add some tests later.

File: crates/hir_ty/src/display.rs [hunk 2]
```diff
@@ -1141,7 +1141,7 @@ impl HirDisplay for Path {
                 write!(f, ">")?;
             }
             (_, PathKind::Plain) => {}
-            (_, PathKind::Abs) => write!(f, "::")?,
+            (_, PathKind::Abs) => write!(f, "")?,
             (_, PathKind::Crate) => write!(f, "crate")?,
```
Comment (lnicola) [inline crates/hir_ty/src/display.rs:1145]: Any idea why this wasn't working?

Comment (dzvon) [reply crates/hir_ty/src/display.rs:1145]: If I still keep `::`, then the absolute path will be displayed in a way that starts with `::::`.

Comment (flodiebold) [reply crates/hir_ty/src/display.rs:1145]: The `write!(f, "")` is unnecessary though, you can just make it `{}`

Comment (dzvon) [reply crates/hir_ty/src/display.rs:1145]: okay

Comment (lnicola) [reply crates/hir_ty/src/display.rs:1145]: No, I meant the line below (`(_, PathKind::Crate) => write!(f, "crate")?`).

Comment (lnicola) [reply crates/hir_ty/src/display.rs:1145]: Never mind, I was a bit confused.

**Comment by dzvon** [comment]
I don't know how to write a testcase for `PathKind::Abs`, seems like `::a` would be interpreted as the crate `a`. 
Any suggestions for me?

**Pull request merged**